ext/opentelemetry-ext-elasticsearch/src/opentelemetry/ext/elasticsearch/__init__.py

```diff
@@ -0,0 +1,155 @@
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""
+This library allows tracing HTTP elasticsearch made by the
+`elasticsearch <https://elasticsearch-py.readthedocs.io/en/master/>`_ library.
+
+Usage
+-----
+
+.. code-block:: python
+
+    import elasticsearch
+    import opentelemetry.ext.elasticsearch
+
+    opentelemetry.ext.elasticsearch.ElasticSearchInstrumentor().instrument()
+    es = elasticsearch.Elasticsearch()
+    es.index(index='my-index', doc_type='my-type', id=1, body={'my': 'data', 'timestamp': datetime.now()})
+    es.get(index='my-index', doc_type='my-type', id=1)
+
+API
+---
+
+# TODO: mention config params
+"""
+
+import functools
+import types
+from logging import getLogger
+from os import environ
+
+import elasticsearch
+import elasticsearch.exceptions
+from wrapt import ObjectProxy
+from wrapt import wrap_function_wrapper as _wrap
+
+from opentelemetry import context, propagators, trace
+from opentelemetry.auto_instrumentation.instrumentor import BaseInstrumentor
+from opentelemetry.ext.elasticsearch.version import __version__
+from opentelemetry.trace import SpanKind, get_tracer
+from opentelemetry.trace.status import Status, StatusCanonicalCode
+
+logger = getLogger(__name__)
+
+
+# Values to add as tags from the actual
+# payload returned by Elasticsearch, if any.
+_attributes_from_result = [
+    "found",
+    "timed_out",
+    "took",
+]
+
+_default_op_name = "request"
+
+
+class ElasticsearchInstrumentor(BaseInstrumentor):
+    """An instrumentor for requests
+    See `BaseInstrumentor`
+    """
+
+    def __init__(self, span_name_prefix=None):
+        if not span_name_prefix:
+            span_name_prefix = environ.get(
+                "OPENTELEMETRY_PYTHON_ELASTICSEARCH_NAME_PREFIX",
+                "Elasticsearch",
+            )
+        self.span_name_prefix = span_name_prefix.strip()
+        super().__init__()
+
+    def _instrument(self, **kwargs):
+        """
+        Instruments elasticsarch module
+        """
+        tracer_provider = kwargs.get("tracer_provider")
+        tracer = get_tracer(__name__, __version__, tracer_provider)
+        _wrap(
+            elasticsearch,
+            "Transport.perform_request",
+            _wrap_perform_request(tracer, self.span_name_prefix),
+        )
+
+    def _uninstrument(self, **kwargs):
+        _unwrap(elasticsearch.Transport, "perform_request")
+
+
+def _wrap_perform_request(tracer, span_name_prefix):
+    def wrapper(wrapped, _, args, kwargs):
+        method = url = None
+        try:
+            method, url = args
+        except IndexError:
+            logger.warning(
+                "expected perform_request to receive two positional arguments. "
+                "Got %d",
+                len(args),
+            )
+
+        op_name = span_name_prefix + (url or method or _default_op_name)
```

Comment:
`span_name_prefix` could still be `None`

Reply:
We set it to a string in __init__ when it is none. It could still be None if a user changed it manually. Renamed the attribute to `_span_name_prefix` to indicate users shouldn't do that.